src/firstvoices/backend/models/characters.py

```diff
@@ -143,21 +103,123 @@ def __str__(self):
         return self.title
 
 
-# See fw-4172: AlphabetMapper model implementation once g2p work has been completed
-# class AlphabetMapper(BaseSiteContentModel):
-#     """
-#     Represents a private table holding the g2p mappings and configuration for a site.
-#     """
-#
-#     class Meta:
-#         verbose_name = _("alphabet mapper")
-#         verbose_name_plural = _("alphabet mappers")
-#
-#     # See fw-4172: Once file storage is configured, enable these FileFields / change to proper paths
-#     # alphabet_mapper is a placeholder for now.
-#     # input_to_canonical = models.FileField(upload_to="alphabet_mapper")
-#     # canonical_to_base = models.FileField(upload_to="alphabet_mapper")
-#     # g2p_config = models.FileField(upload_to="alphabet_mapper")
-#
-#     def __str__(self):
-#         return self.id
+class AlphabetMapper(BaseSiteContentModel):
+    """
+    Hosts text processors and sorters for a site's custom alphabet.
+    Must be re-created upon changing any character models in an alphabet.
```

Comment:
I'm sure we've talked about this before so I'm sorry for asking, but why does this need to be re-created after changing any character models? It looks like the only thing that is stored is the input_to_canonical_map. If I added a character or changed the characters' sort orders, wouldn't this just keep working with the existing confusable list?

Reply:
Yes you are correct, I will update the comment. The only thing that needs manual adjustment is after an alphabet is edited the confusables map on the alphabet mapper should be updated to include any confusable characters.